lib/directory-tree.js

```diff
@@ -25,17 +25,26 @@ function directoryTree (path, options, onEachFile) {
 	const item = { path, name };
 	let stats;
 
+    let depth = 1;
```

Comment:
The whitespace is still misaligned somehow, compare this line with line 26 above.

Reply:
Why do you need this `depth` var? It's not helpful in any way. Basically, you just need to keep decrementing options.depth until you reach 0 or < 0, your call. Let's just stick to not specifying depth == full depth. No need for the `-1` trick. It will simplify the code.

Furthermore, what I was saying previously is that you should be checking for depth at the beginning of the function, not inside the if for Directory.

Reply:
If I'd check the depth at the beginning of the function and the maximum depth is reached just { path, name } would be returned for files instead of the full information.